python/ray/_private/node.py

```diff
@@ -317,6 +324,41 @@ def __init__(
         self.validate_ip_port(self.gcs_address)
         self._record_stats()
 
+    def check_persisted_session_name(self):
+        if self._ray_params.external_addresses is None:
```

Comment:
Could you please reuse this part for address parsing

https://github.com/ray-project/ray/blob/master/python/ray/_private/services.py#L1338-L1349

Reply:
Seems like copy paste.